src/lib/routes/auth/simple-password-provider.test.js

```diff
@@ -0,0 +1,83 @@
+const test = require('ava');
+const request = require('supertest');
+const express = require('express');
+const User = require('../../user');
+const PasswordProvider = require('./simple-password-provider');
+
+const getLogger = () => ({ info: () => {}, error: () => {} });
+
+test('Should require password', async t => {
+    const app = express();
+    app.use(express.json());
+    const userService = () => {};
+    const ctr = new PasswordProvider({ getLogger }, { userService });
+
+    app.use('/auth/simple', ctr.router);
+
+    const res = await request(app)
+        .post('/auth/simple/login')
+        .send({ name: 'john' });
+
+    t.is(400, res.status);
```

Comment:
This should also be 401 I would think

Reply:
I think 400 Bad Request is correct here, because the request body is missing the required password parameter. 401 gives you information about your login credentials being valid given that you have provided the correct details. At least that's how I see it.